packages/shared/src/contexts/AuthContext.tsx

```diff
@@ -120,7 +122,29 @@ export const AuthContextProvider = ({
 }: AuthContextProviderProps): ReactElement => {
   const [loginState, setLoginState] = useState<LoginState | null>(null);
   const endUser = user && 'providers' in user ? user : null;
-  const referral = user?.referrer;
+  const referralCampaign = useMemo<{
+    referralId: string;
+    referralOrigin: string;
+  }>(() => {
+    const cookie = typeof window !== 'undefined' ? getCookieObject() : {};
```

Comment:
Wouldn't the window will always be undefined at the beginning in the webapp?

Reply:
It would be during SSR, but on first render of the app window would be so the context would initialise with correct value.